test/system/projects/samples_test.rb

```diff
@@ -41,1291 +199,1535 @@ class SamplesTest < ApplicationSystemTestCase
-        assert_text @sample2.puid
-      end
+      # remaining samples still in table
+      assert_text strip_tags(I18n.t(:'viral.pagy.limit_component.summary', from: 1, to: 2, count: 2,
+                                                                           locale: @user.locale))
+      ### VERIFY END ###
     end
 
-    test 'should transfer multiple samples' do
-      project2 = projects(:project2)
+    test 'transfer dialog sample listing' do
+      ### SETUP START ###
+      samples = @project.samples.pluck(:puid, :name)
       visit namespace_project_samples_url(@namespace, @project)
-      assert_text strip_tags(I18n.t(:'viral.pagy.limit_component.summary', from: 1, to: 3, count: 3,
-                                                                           locale: @user.locale))
-      within '#samples-table table tbody' do
-        assert_selector 'tr', count: 3
-        all('input[type=checkbox]').each { |checkbox| checkbox.click unless checkbox.checked? }
+      ### SETUP END ###
+
+      ### ACTIONS START ###
+      click_button I18n.t(:'projects.samples.index.select_all_button')
+      click_link I18n.t('projects.samples.index.transfer_button')
+      ### ACTIONS END ###
+
+      ### VERIFY START ###
+      within('#list_selections') do
+        samples.each do |sample|
+          assert_text sample[0]
+          assert_text sample[1]
+        end
       end
-      click_link I18n.t('projects.samples.index.transfer_button'), match: :first
-      within('span[data-controller-connected="true"] dialog') do
+      ### VERIFY END ###
+    end
+
+    test 'transfer dialog with plural description' do
+      ### SETUP START ###
+      visit namespace_project_samples_url(@namespace, @project)
+      ### SETUP END ###
+
+      ### ACTIONS START ###
+      click_button I18n.t(:'projects.samples.index.select_all_button')
+      click_link I18n.t('projects.samples.index.transfer_button')
+      ### ACTIONS END ###
+
+      ### VERIFY START ###
+      within('#dialog') do
         assert_text I18n.t('projects.samples.transfers.dialog.description.plural').gsub!('COUNT_PLACEHOLDER',
                                                                                          '3')
-        within %(turbo-frame[id="list_selections"]) do
-          samples = @project.samples.pluck(:puid, :name)
-          samples.each do |sample|
-            assert_text sample[0]
-            assert_text sample[1]
-          end
-        end
-        find('input#select2-input').click
-        find("button[data-viral--select2-primary-param='#{project2.full_path}']").click
-        click_on I18n.t('projects.samples.transfers.dialog.submit_button')
       end
+      ### VERIFY END ###
     end
 
-    test 'should transfer a single sample' do
-      project2 = projects(:project2)
+    test 'transfer dialog with singular description' do
+      ### SETUP START ###
       visit namespace_project_samples_url(@namespace, @project)
```

Comment:
**nitpick (blocking):** Same as comment above.

Reply:
Added in d60b9028eeca99d5aad54b37f1e7ce872ceb8dac